### PR TITLE
Create system 0.3.0 package

### PR DESCRIPTION
**Pull request by fearful-symmetry**
This was a tad fraught since there's enough changes between master and experimental that I couldn't just cherry-pick the PR. There's a few outstanding issues:

- I'm still running into this: https://github.com/elastic/kibana/issues/68397
- And this: https://github.com/elastic/kibana/issues/68147
- This will come after this PR: https://github.com/elastic/integrations/issues/26
- I'm now running into an issue where the maps aren't being installed, and I'm not sure if it's something I'm doing or not. (known issue, not me)
- The README file is missing an entry on `log` fields

If you want to test this, here's a docker-compose file:
```
# This should start the environment with the latest snapshots.

version: '2.3'
services:
  elasticsearch:
    image: docker.elastic.co/elasticsearch/elasticsearch:7.8.0-SNAPSHOT
    healthcheck:
      test: ["CMD", "curl", "-f", "-u", "elastic:changeme", "http://127.0.0.1:9200/"]
      retries: 300
      interval: 1s
    environment:
    - "ES_JAVA_OPTS=-Xms1g -Xmx1g"
    - "network.host="
    - "transport.host=127.0.0.1"
    - "http.host=0.0.0.0"
    - "indices.id_field_data.enabled=true"
    - "xpack.license.self_generated.type=trial"
    - "xpack.security.enabled=true"
    - "xpack.security.authc.api_key.enabled=true"
    - "ELASTIC_PASSWORD=changeme"
    ports:
      - "0.0.0.0:9200:9200"

  kibana:
    image: docker.elastic.co/kibana/kibana:7.8.0-SNAPSHOT
    depends_on:
      elasticsearch:
        condition: service_healthy
      package-registry:
        condition: service_healthy
    healthcheck:
      test: "curl -f http://127.0.0.1:5601/login | grep kbn-injected-metadata 2>&1 >/dev/null"
      retries: 600
      interval: 1s
    volumes:
      - ./kibana.config.yml:/usr/share/kibana/config/kibana.yml
    ports:
      - "0.0.0.0:5601:5601"

  package-registry:
    image: docker.elastic.co/package-registry/package-registry:experimental
    healthcheck:
      test: ["CMD", "curl", "-f", "http://127.0.0.1:8080"]
      retries: 300
      interval: 1s
    volumes:
      - ../build/public:/registry/public
    ports:
      - "0.0.0.0:8080:8080"
```

You'll need this as well:
```
server.name: kibana
server.host: "0"

elasticsearch.hosts: [ "http://elasticsearch:9200" ]
elasticsearch.username: elastic
elasticsearch.password: changeme
xpack.monitoring.ui.container.elasticsearch.enabled: true

xpack.ingestManager.enabled: true
xpack.ingestManager.epm.enabled: true
xpack.ingestManager.epm.registryUrl: "http://package-registry:8080"
xpack.ingestManager.fleet.enabled: true
xpack.ingestManager.fleet.elasticsearch.host: "http://localhost:9200"
xpack.ingestManager.fleet.kibana.host: "http://localhost:5601"
xpack.ingestManager.fleet.tlsCheckDisabled: true
``` 

Then just run `docker-compose -f snapshot.yml pull` then `docker-compose -f snapshot.yml -f local.yml up --force-recreate`

### CODE REVIEW

**Comment by mtojek** [comment]
Re: https://github.com/elastic/kibana/issues/68397 :

It's because the branch is not in sync with master, when the issue has been fixed. See differences:
https://github.com/elastic/package-storage/blob/master/packages/system/0.3.0/dataset/cpu/agent/stream/stream.yml.hbs
vs
https://github.com/fearful-symmetry/package-storage/blob/system-03-exp/packages/system/0.3.0/dataset/cpu/agent/stream/stream.yml.hbs

I assume that the issue doesn't exist on master, right?

Re: https://github.com/elastic/kibana/issues/68147

I think we talked about this issue last week and we're informed by the Kibana team that they will fix routing to bring back support for legacy anchors. If you prefer to fix the navigation immediately, you need to update all anchors to the modern scheme.

**Comment by ruflin** [comment]
@mtojek For https://github.com/elastic/kibana/issues/68147 to you know if this issue also exists in 7.8? Because this one here is only for 7.8.

**Comment by ruflin** [comment]
To simplify my own testing I added the testing files locally but also opened this PR here: https://github.com/elastic/package-storage/pull/79 I suggest we get this in to make testing for others easier and rebase / merge it into this PR here.

**Comment by mtojek** [comment]
> To simplify my own testing I added the testing files locally but also opened this PR here: #79 I suggest we get this in to make testing for others easier and rebase / merge it into this PR here.

PR #79 approved, but I assume this another call for defining a flow between experimental and master :)

**Comment by mtojek** [comment]
> @mtojek For [elastic/kibana#68147](https://github.com/elastic/kibana/issues/68147) to you know if this issue also exists in 7.8? Because this one here is only for 7.8.

I think it's only for 8.0.0-SNAPSHOT, but I've asked the question in the relevant thread:

https://github.com/elastic/kibana/issues/68147#issuecomment-645198845

**Comment by ycombinator** [comment]
I'm currently testing on Mac OS X. In the `[Metrics System] Overview ECS` dashboards, the `Disk used [Metrics System] ECS` visualization shows 0%. I tried changing the time picker from "Last 15 minutes" to "Last 30 minutes" or "Last 1 hour" but the visualization still shows 0%.

Screen Shot 2020-06-17 at 9 06 47 AM

I checked the metricsets reporting data via this query:

```
POST metrics-*/_search?filter_path=aggregations
{
  "size": 0, 
  "query": {
    "bool": {
      "filter": {
        "term": {
          "service.type": "system"
        }
      }
    }
  },
  "aggs": {
    "by_metricset": {
      "terms": {
        "field": "metricset.name",
        "size": 20
      }
    }
  }
}
```

And the result was:

```
{
  "aggregations" : {
    "by_metricset" : {
      "doc_count_error_upper_bound" : 0,
      "sum_other_doc_count" : 0,
      "buckets" : [
        {
          "key" : "network",
          "doc_count" : 4332
        },
        {
          "key" : "process",
          "doc_count" : 1695
        },
        {
          "key" : "diskio",
          "doc_count" : 228
        },
        {
          "key" : "load",
          "doc_count" : 228
        },
        {
          "key" : "memory",
          "doc_count" : 228
        },
        {
          "key" : "process_summary",
          "doc_count" : 228
        },
        {
          "key" : "socket_summary",
          "doc_count" : 228
        },
        {
          "key" : "uptime",
          "doc_count" : 228
        }
      ]
    }
  }
}
```

**Comment by ycombinator** [comment]
Probably related to the previous comment but in the `[Metrics System] Host overview ECS` dashboard, `Disk used [Metrics System] ECS` shows 0% and `Disk Usage [Metrics System] ECS` shows `No data to display for the selected metrics`.

Screen Shot 2020-06-17 at 9 44 31 AM

**Comment by ycombinator** [comment]
On the `[Metrics System] Host Services Overview` dashboard the top row of visualizations all show `Could not locate that index-pattern-field` errors. I tried refreshing the index pattern but it didn't seem to help. 

The bottom two rows of visualizations show `No data to display for the selected metrics`.

Screen Shot 2020-06-17 at 9 47 49 AM

Note that I'm testing on Mac OS, if that matters.

**Comment by ycombinator** [comment]
The visualizations in the `[Metrics System] Containers overview ECS` are completely empty but I think that's expected because I'm not on a Docker host. I will test this again by running Agent on a Docker host.

**Comment by fearful-symmetry** [comment]
So, fixed the disk gauge issue. The system dashboard is empty as it's a linux-only metricset.

File: packages/system/0.3.0/docs/README.md [hunk 1]
```diff
@@ -0,0 +1,618 @@
...
+https://gitlab.com/apparmor/apparmor/wikis/TechnicalDoc_Proc_and_ptrace[AppArmor
+and other LSM software], even though the System module doesn't use `ptrace`
+directly.
+
+## Compatibility
+
+The System metricsets collect different kinds of metric data, which may require dedicated permissions
+to be fetched and which may vary across operating systems.
+
+## Metrics
```
Comment (kaiyan-sheng) [inline packages/system/0.3.0/docs/README.md:21]: Seems like we only have `## Metrics` session here but not for logs?

Comment (fearful-symmetry) [reply packages/system/0.3.0/docs/README.md:21]: All of that is auto-generated by...something? Not sure.

**Comment by kaiyan-sheng** [comment]
Same issue with kibana index pattern happened when testing syslog dashboard. But after manually refresh the index pattern, the dashboard looks good:
Screen Shot 2020-06-17 at 11 19 49 AM

One problem about system logs, I don't see documentation around that in:
Screen Shot 2020-06-17 at 11 25 09 AM

This can be a separate PR, I will look into it later.

**Comment by fearful-symmetry** [comment]
The, the index issues are known and unrelated to this.

File: packages/system/0.3.0/dataset/auth/fields/package-fields.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+- name: system
```
Comment (ruflin) [inline packages/system/0.3.0/dataset/auth/fields/package-fields.yml:1]: @neptunian Are these files causing problems potentially?

**Pull request merged**